r-package/survlearners/R/surv_fl_grf.R

```diff
@@ -22,19 +23,17 @@
 #' censor.time <- (numeratorC/(4^2))^(1/2)
 #' Y <- pmin(failure.time, censor.time)
 #' D <- as.integer(failure.time <= censor.time)
-#' data <- list(X = X, W = W, Y = Y, D = D)
-#' data.test <- list(X = X, W = W, Y = Y, D = D)
 #'
-#' cate = surv_fl_grf(data, data.test, times, ps = 0.5)
+#' cate = surv_fl_grf(X, W, Y, D, times, ps = 0.5, newX = X)
 #' }
 #' @return A vector of estimated conditional average treatment effects
 #' @export
-surv_fl_grf <- function(data, data.test, times, alpha = 0.05, ps = NULL, cen_fit = "KM"){
+surv_fl_grf <- function(X, W, Y, D, times, alpha = 0.05, ps = NULL, cen_fit = "KM", newX = NULL){
```

Comment:
It should be: (X, Y, W, D) to be consistent with grf

there should be no `newX` argument here

predict.surv_fl_grf should have a `newdata` argument, not `newX`

Reply:
i.e. you need to add predict.surv* for all of them

Reply:
re consistent with grf: I got it... it happened due to some file conflicts earlier. I am working on resolving it 

For the newX, I am building a function so that users can specify on which X to make predictions, can be the original X or new X. I think it is more about how to design the functions here, I do this so that I don't need predict.surv_*, but what you suggested is definitely another approach

Reply:
The surv_ function should look like the rest of the estimators and conform to what is standard for almost every other package:

model = surv_***(X, Y, W, D, etc.arguments) fits a surv_**  model using data X,Y,W,D

predict(mode) should give (OOB/cross fit if available) predictions at the training data  X

predict(model, X.test) should give predictions at new test data X

Reply:
You are right on being standard, I agree, thanks for pointing it out! I will make that adjustment

Reply:
Great! Remember to practice splitting up your changes in good PRs, you can write down a plan of what you have to do:

a)
b)
c)
etc

and try to think ahead that c) and be done cleanly after b)

again the plan is in the futre you will have a main log here:
<img width="560" alt="Screen Shot 2022-04-10 at 16 31 57" src="https://user-images.githubusercontent.com/7185264/162644650-ee563ae2-faa3-41c2-9b1d-5058cb2af897.png">


that tells you everything you need to know what youve done and when. this will be important for for ex the paper:

if it says

-1"Remove incosistent fomatting"
-2"Change estimator argument order"

you KNOW that if you reran your HPC simulations and got different results it should NOT be because of "1" because that made ONLY aesthetic changes, so must have been b/c of 2" or something before ... !

Reply:
Got it! thanks for emphasizing its importance. Sometimes, I have to catch/remind myself to be more organized instead of trying to fix everything in one commit (that tendency gets strong when there is a lot to do, haha). I will practice and try to improve~~

Reply:
yes, remember to resist the urge to do everything in one big thing

splitting things up often means extra work as in this setting you'll likely touch the same  lines twice++

however it reduces the change of _mistakes_ since if you split things up you touch that same line for DIFFERENT reasons in separate commits. if everything is done at once those two changes to the same line is done at once and you are less likely to know if change a) or change b) caused an error